src/Riak/Command/Indexes/Query.php

```diff
@@ -0,0 +1,204 @@
+            $this->lowerBound = $builder->getLowerBound();
+            $this->upperBound = $builder->getUpperBound();
+        }
+        else {
+            $this->match = $builder->getMatchValue();
+        }
+
+        $this->continuation = $builder->getContinuation();
+        $this->maxResults = $builder->getMaxResults();
+        $this->returnTerms = $builder->getReturnTerms();
+        $this->paginationSort = $builder->getPaginationSort();
+        $this->termFilter = $builder->getTermFilter();
+        $this->timeout = $builder->getTimeout();
+    }
+
+    public function getIndexName() {
+        return $this->indexName;
+    }
+
+    public function getMatchValue() {
+        return $this->match;
+    }
+
+    public function getLowerBound() {
+        return $this->lowerBound;
+    }
+
+    public function getUpperBound() {
+        return $this->upperBound;
+    }
+
+    public function isMatchQuery()
+    {
+        return isset($this->match);
+    }
+
+    public function isRangeQuery()
+    {
+        return isset($this->lowerBound) && isset($this->upperBound);
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getContinuation()
+    {
+        return $this->continuation;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getMaxResults()
+    {
+        return $this->maxResults;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function getReturnTerms()
+    {
+        return $this->returnTerms;
+    }
+
+    /**
+     * @return bool|null
+     */
+    public function getPaginationSort()
+    {
+        return $this->paginationSort;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getTermFilter()
+    {
+        return $this->termFilter;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getTimeout()
+    {
+        return $this->timeout;
+    }
+
+        /**
+     * @return Command\Indexes\Response
+     */
+    public function getResponse()
+    {
+        return $this->response;
+    }
+
+    public function getParameters()
+    {
+        $parameters = [];
+
+        if(isset($this->continuation)) {
+            $parameters['continuation'] = $this->continuation;
+        }
+
+        if(isset($this->maxResults)) {
+            $parameters['max_results'] = $this->maxResults;
+        }
+
+        if(isset($this->returnTerms)) {
+            $parameters['return_terms'] = ($this->returnTerms) ? 'true' : 'false';
+        }
+
+        if(isset($this->paginationSort)) {
+            $parameters['pagination_sort'] = ($this->paginationSort) ? 'true' : 'false';
+        }
+
+        if(isset($this->termFilter)) {
+            $parameters['term_regex'] = $this->termFilter;
+        }
+
+        if(isset($this->timeout)) {
+            $parameters['timeout'] = $this->timeout;
+        }
+
+        return $parameters;
+    }
+
+    public function hasParameters()
+    {
+        return true;
+    }
```

Comment:
Hey, @MarcoDeBortoli. Yea, hasParameters is in the CommandInterface because of it's use within the API Bridge layer to determine if parameters need to be appended to the request